lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +681,159 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    private final StatusEvent updateEventForDeletion(final StatusEvent ev)
+              throws RetriableIndexingException {
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get()).longValue();
+
+        Map<String, Object> command;
+
+        // check if workspace is permanently deleted or marked as deleted
+        try {
+            final Tuple9 tuple = getWorkspaceInfo(wsid);
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            // hacky: if exception message is changed, this logic will silently fail
+            if (ex.getMessage().contains("No workspace with id ")) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
```

Comment:
If I use the other constructor that takes storageCode as a param, the code crashes on the indexObject method since storageObjectType is a required parameter there. So I am not sure if that is a bug. If it is, that would be a bigger fix to make.

Reply:
indexObject only takes NEW_VERSION events though. The other event types go to other methods where a StorageObjectType is not expected - and won't be present in the event.

Reply:
this is another case where some better documentation and state checking code in the StatusEvent class would be helpful, I think.

Reply:
Since we're not going to talk today but I'm reviewing, some more details about this:
Only NEW_VERSION events will have a StorageObjectType. All the other events will just have a storage code. This makes sense because all the other events affect *objects* rather than *object versions*. Since versions of the same object do not need to have the same type, the event cannot have a StorageObjectType. NEW_VERSION events, in contrast, *always* have a StorageObjectType. The StorageObjectType is only accessed in the indexObjects() function in the worker, where only NEW_VERSION events will be dispatched.

Again, some docs/constraints would help here. I keep thinking that maybe we should split the event class into subclasses for each type but that might make things worse.

Reply:
"Since versions of the same object do not need to have the same type, the event cannot have a StorageObjectType." This I find very counter-intuitive and keep struggling with. Anyway, I have added your clarification above to the StatusEvent documentation and updated code to use storage code when storage object type is not present.